app/controllers/registrations_controller.rb

```diff
@@ -822,19 +823,69 @@ def create
   end
 
   def create_paypal_order
-    return head :forbidden if Rails.env.production?
+    return head :forbidden if PaypalInterface.paypal_disabled?
 
     @registration = registration_from_params
-    render json: PaypalInterface.create_order(@registration)
+    render json: PaypalInterface.create_order(@registration, params[:total_charge])
   end
 
   def capture_paypal_payment
-    return head :forbidden if Rails.env.production?
+    return head :forbidden if PaypalInterface.paypal_disabled?
 
     @registration = registration_from_params
     @competition = @registration.competition
     order_id = params[:order_id]
 
-    render json: PaypalInterface.capture_payment(@competition, order_id)
+    response = PaypalInterface.capture_payment(@competition, order_id)
+    if response['status'] == 'COMPLETED'
+
+      # TODO: Handle the case where there are multiple captures for a payment
+      # 1) Multiple installments
+      # 2) Some failed, some succeeded
+
+      amount_details = response['purchase_units'][0]['payments']['captures'][0]['amount']
+      currency_code = amount_details['currency_code']
+      amount = PaypalRecord.amount_in_cents(amount_details["value"], currency_code)
+      record = PaypalRecord.find_by(record_id: response["id"])
+
+      # Create a Capture object and link it to the PaypalRecord
+      # NOTE: This assumes there is only ONE capture per order - not a valid long-term assumption
+      capture_from_response = response['purchase_units'][0]['payments']['captures'][0]
+      PaypalRecord.create(
+        record_id: capture_from_response['id'],
+        status: capture_from_response['status'],
+        payload: {}, # TODO: Refactor so that we can actually capture the payload? Perhaps this needs to be called in PaypalInterface?
+        amount_in_cents: capture_from_response['amount']['value'],
+        currency_code: capture_from_response['amount']['currency_code'],
+        record_type: :capture,
+        parent_record: record,
+      )
+
+      # Record the payment
+      @registration.record_payment(
+        amount,
+        currency_code,
+        record, # TODO: Add error handling for the PaypalRecord not being found
+        @registration.user.id,
+      )
+    end
+
+    render json: response
+  end
+
+  def refund_paypal_payment
+    registration = Registration.find(params[:id])
+    paypal_order = RegistrationPayment.find(params[:payment_id]).receipt
+    payment_capture_id = paypal_order.capture_id
+
+    refund = PaypalInterface.issue_refund(registration, payment_capture_id)
+
+    registration.record_refund(
+      refund.amount_in_cents.to_i,
```

Comment:
Also, should I make the wording consistent with stripe_transaction - `amount_lowest_denomination`? I like `amount_in_cents` because it makes it very obvious* what is actually being stored ("denomination" is not a word I have an instant undertanding of, but it is obviously more technically correct)

*to me because currencies I'm familiar with use cents

Reply:
It's not necessarily technically correct. It's only what Philippe(?) happened to choose at the time.
Actually, we're trying to avoid the `_lowest_denomination` suffix in another WCIF discussion actively because it's too long and a mouthful.

The essential point here is not to find good financial terminology, but rather to distinguish between "the amount in a form that Paypal wants" and "the amount in a form that Ruby needs to write to our database". I think the conversion methods in Stripe are called "def amount_to_stripe(foo)" and "def amount_to_ruby(foo)". That's the point here.

Reply:
Ok thanks for the insight! I've made the following changes:
- Changed method name from `amount_in_cents` to `ruby_amount`
- Kept name of database field the same (`amount_in_cents`) - both to reduce confusion between the method and the property, and to explicitly state what is being stored in the db